text/0134-remove-span-status.md

```diff
@@ -0,0 +1,88 @@
+
+## Internal details
+
+Several areas are affected by this change and will require the coordinate effort to make this happen.
+
+### Trace API Specification
+
+* Remove all mentions of `Status` and `Set Status` API from Span section.
+* Remove whole `Status` section.
+
+### Trace SDK Specification
+
+No changes needed.
+
+### Trace SDK exporters
+
+Remove translation rules pertaining to `Span.Status`.
+
+### Span semantic conventions
+
+* From HTTP conventions, remove `Status` section.
+* In gRPC conventions, replace the requirement to set span status with the requirement to set span attribute `grpc.status`
+to the canonical status code of the gRPC response.
+
+### OTLP data definition
+
+Rename `Span.status` field to `Span.deprecated_status` field.
+
+### Language API and SDK
+
+* Every language must remove status related code from their implementations of Trace API and SDK.
+* Every language must remove all code in their span exporters which serializes span status.
+* All instrumentation libraries, both manual and automated, must remove all calls to the removed `Span.setStatus` API.
+* All instrumentation libraries, both manual and automated, must adapt new semantic conventions for gRPC spans.
+
+### Collector
+
+Collector will have a transition period during which it has a special handling both for old data,
+which has `Span.Status` present, and new data without Status.
+This will allow for older pipelines, still relying on `Span.Status` to continue to work with as little disruption as possible.
+The exact duration of this transition period will be left for Collector's maintainers to decide.
+
+* If incoming OTLP data has `Span.Status` present, that will NOT be removed and will be passed on as-is.
+* If incoming OTLP data has `Span.Status` present, Collector will translate that to temporary semantic attributes as follows:
+
+|Status|Tag Key| Tag Value|
+|--|--|--|
+|StatusCanonicalCode | `otel.deprecated_status_code` | Name of the code, if that differs from ‘OK’ |
+|Message *(optional)* | `otel.deprecated_status_description` | `{message}` |
+
+* When receiving data in Zipkin format, no special handling will take place.
+All Zipkin tags present will be translated verbatim to OTLP attributes.
+* When exporting data into Zipkin format, no special handling will take place.
+All OTLP attributes will be translated verbatim to Zipkin tags.
+`Span.Status` if present, will NOT be translated.
+* When receiving data in Jaeger format, no special handling will take place.
+Jaeger `error` tag will be translated in the usual way to OTLP `error` attribute.
+See `Open questions` below.
+* When exporting data into Jaeger format, if a new temporary attribute `otel.deprecated_status_code` is present,
+then Jaeger `error` tag will be set to `true`.
+See `Open questions` below.
+
+## Open questions
+
+Neither OpenTelemetry Specification nor Protocol currently assigns any special meaning to the `error` attribute mentioned above
```

Comment:
I am not quite sure what happens if Status is removed but we do not (yet) have the "error"-like attribute. How will the errors be indicated? Do we lose this capability? Or the intent is to wait until "error" indicator (whatever it is) is first added to the specification and only after that we remove Status?

Reply:
Current proposal is to remove status right now, file a follow up issue and work with Errors WG to resolve it. Either via one of the already existing proposals, like open-telemetry/opentelemetry-specification#822, or a new one.

Reply:
Got it. I see in this OTEP that you already thought through the transition process, which is great, but I need to mention this for visibility: we must be careful with Collector changes. There are people who use OpenTelemetry Collector in production already. All intermediate states of the Collector need to be fully functional.

Reply:
@tigrannajaryan It seems open-telemetry/opentelemetry-specification#822 will be effectively merged soon (the `error.hint` Span attribute at least). Would that be enough for the Collector?

Reply:
I am reluctant to give an approval until I clearly understand what we replace this by.

If https://github.com/open-telemetry/opentelemetry-specification/pull/822 is the replacement can we wait for it be merged first and update this OTEP to make use of it? It is an open question in this OTEP and is one of the central problems that this OTEPs needs to solve, but remains partially unsolved for now.

Reply:
@iNikem To be clear, I am not opposed to this OTEP, but I think it could be improved and an open question can be hopefully closed if https://github.com/open-telemetry/opentelemetry-specification/pull/822 is accepted.

We do not typically alter OTEPs after they are merged, so I don't know if a follow-up PR that would make the improvement is a reasonable approach. Because of this I also think that it may make sense to wait for https://github.com/open-telemetry/opentelemetry-specification/pull/822 and then make any required changes to this OTEP.

If you do not think that https://github.com/open-telemetry/opentelemetry-specification/pull/822 can have any significant impact on this OTEP then we don't need to wait for it and can assume https://github.com/open-telemetry/opentelemetry-specification/pull/822 does not exist. However in that case the way I perceive this PR changes a bit, because it seems to leave us without standardized error reporting approach and I think we need such approach.

Reply:
I think the otep may change as a result of resolving any open question, but that doesn't mean we should stop this. I think the main point of this otep is the removing of span status and an action item is required to think about if we need an equivalent of `error.hint` or not separately.

Trying to make progress so we don't get stuck, but I can see your point as well, if you suggest that we really need a way to signal error from the application (which I don't believe we need) then I understand why we should block this until we have that.

Reply:
@tigrannajaryan can you express your opinion in open-telemetry/opentelemetry-specification#822? If you think that it may resolve your hesitation about this PR

Reply:
> I am not against moving forward with this OTEP, but I believe it will have an important open question and will be unactionable in the Collector if it does not take into account either the existence of "error.hint" or the lack of whatever else needs to replace Span Status. IMO, it will mean we will need another OTEP to correct this OTEP once "error.hint" is merged or rejected.

Not sure you need an OTEP for that, that is a semantic convention, and collector will use it.

Reply:
@bogdandrutu sure, that probably works too.